tests/openssh-mux-client.rs

```diff
@@ -0,0 +1,805 @@
+#![cfg(feature = "mux_client")]
```

Comment:
I feel like for _most_ of the test suite it makes sense to run each test twice — once using `mux` and once using `process` — rather than copy-pasting the exact same test with just a _slight_ change to the connect mechanism. That will also make it easier to update tests over time, and to add new ones that cover both impls. You should be able to make a pretty decent attempt at it with a simple `macro_rules!` I think.

Reply:
Yeah, most of them are indeed the same and only a few is different.

I will merge these duplicated into one file

Reply:
All of these tests except for `process_exit_on_signal` is merged.

`process_exit_on_signal` cannot be merged since `process_impl` and `mux_client_impl` has different behaviors when the remote process are killed by a signal.

Currently `mux_client_impl` returns a dummy status code, however this is not correct, so I would instead create a new `Error` variant for this situation.

Reply:
It's done, now they are merged back into one file `tests/openssh.rs`.